worktree.go

```diff
@@ -897,6 +900,130 @@ func (w *Worktree) Submodule(name string) (*Submodule, error) {
+}
+
+// resolveHTTPRelativeURL resolves relative URLs for HTTP/HTTPS protocols
+func (w *Worktree) resolveHTTPRelativeURL(originURL, relativeURL string) (string, error) {
+	// For HTTPS/HTTP URLs like https://github.com/user/repo.git
+	base, err := url.Parse(originURL)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove .git suffix if present to get the directory path
+	base.Path = strings.TrimSuffix(base.Path, ".git")
+
+	// Ensure the path ends with a slash for proper relative resolution
+	if !strings.HasSuffix(base.Path, "/") {
+		base.Path += "/"
+	}
+
+	// Resolve relative path
+	resolved, err := url.Parse(relativeURL)
+	if err != nil {
+		return "", err
+	}
+
+	resolvedURL := base.ResolveReference(resolved)
+	resolvedURL.Path = strings.TrimSuffix(resolvedURL.Path, "/")
+
+	return resolvedURL.String(), nil
+}
+
+// resolveSSHRelativeURL resolves relative URLs for SSH protocol
+func (w *Worktree) resolveSSHRelativeURL(baseEndpoint *transport.Endpoint, relativeURL, originURL string) (string, error) {
+	// For SSH URLs like git@github.com:user/repo.git
+	if baseEndpoint.Host == "" || baseEndpoint.Path == "" {
+		return "", fmt.Errorf("invalid SSH URL format for origin: %s", originURL)
+	}
+
+	basePath := strings.TrimSuffix(baseEndpoint.Path, ".git")
+	pathParts := strings.Split(basePath, "/")
+
+	relativeParts := strings.Split(relativeURL, "/")
+	for _, part := range relativeParts {
+		if part == ".." && len(pathParts) > 0 {
+			pathParts = pathParts[:len(pathParts)-1]
+		} else if part != "." && part != "" {
+			pathParts = append(pathParts, part)
+		}
+	}
+
+	newPath := strings.Join(pathParts, "/")
+
+	if baseEndpoint.User != "" {
+		return fmt.Sprintf("%s@%s:%s", baseEndpoint.User, baseEndpoint.Host, newPath), nil
+	}
+	return fmt.Sprintf("%s:%s", baseEndpoint.Host, newPath), nil
+}
+
+// resolveFileRelativeURL resolves relative URLs for file:// protocol
+func (w *Worktree) resolveFileRelativeURL(baseEndpoint *transport.Endpoint, relativeURL string) (string, error) {
+	// For file:// URLs like file:///path/to/repo.git
+	if !strings.HasSuffix(relativeURL, ".git") {
+		return "", fmt.Errorf("file:// protocol requires .git suffix, got: %s", relativeURL)
+	}
+
+	baseDir := path.Dir(baseEndpoint.Path)
+	resolvedPath := path.Join(baseDir, relativeURL)
+
+	return fmt.Sprintf("file://%s", resolvedPath), nil
+}
+
+// getOriginRemoteURL extracts the origin remote URL for reuse
+func (w *Worktree) getOriginRemoteURL() (string, error) {
+	remotes, err := w.r.Remotes()
+	if err != nil {
+		return "", err
+	}
+
+	var originRemote *Remote
+	for _, remote := range remotes {
+		if remote.c.Name == "origin" {
+			originRemote = remote
+			break
+		}
+	}
+
+	if originRemote == nil {
+		return "", fmt.Errorf("no origin remote found")
+	}
+
+	if len(originRemote.c.URLs) == 0 {
+		return "", fmt.Errorf("origin remote has no URLs configured")
+	}
+
+	return originRemote.c.URLs[0], nil
+}
```

Comment:
Let's inline this instead - please refer to suggestions on `Submodules()`.
```suggestion
```

Reply:
Fixed via 6d6339a3 .